lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -152,17 +169,28 @@ def decrypt_ejson(key_dir)
 
     def fetch_private_key_from_secret
       KubernetesDeploy.logger.info("Fetching ejson private key from secret #{EJSON_KEYS_SECRET}")
-      secret = @kubeclient.get_secret(EJSON_KEYS_SECRET, @namespace)
+
+      secret = run_kubectl_json("get", "secret", EJSON_KEYS_SECRET)
       encoded_private_key = secret["data"][public_key]
       unless encoded_private_key
         raise EjsonSecretError, "Private key for #{public_key} not found in #{EJSON_KEYS_SECRET} secret"
       end
 
       Base64.decode64(encoded_private_key)
-    rescue KubeException => error
-      raise unless error.error_code == 404
-      secret_missing_err = "Failed to decrypt ejson: secret #{EJSON_KEYS_SECRET} not found in namespace #{@namespace}."
-      raise EjsonSecretError, secret_missing_err
+    end
+
+    def run_kubectl_json(*args)
+      args += ["--output=json"]
+      out, err, st = run_kubectl(*args)
+      raise EjsonSecretError, err unless st.success?
+      result = JSON.parse(out)
+      result.fetch('items', result)
+    end
+
+    def run_kubectl(*args)
+      raise FatalDeploymentError, "Namespace missing for namespaced command" if @namespace.blank?
```

Comment:
Because this is copied from another place where it wouldn't make sense in the constructor. :P

Really, it's a super-paranoid check to make sure we don't run commands without a context/namespace. In reality, it should be impossible to hit in production (several other things would have raised before we get here). But I don't feel strongly about where it goes and can move it.